pkg/list/list_stacks.go

```diff
@@ -1,43 +1,374 @@
+			}
+		}
+
+		// Set appropriate delimiter based on format
+		fileDelimiter := delimiter
+		if delimiter == "\t" || delimiter == "" {
+			switch format {
+			case FormatCSV:
+				fileDelimiter = ","
+			case FormatTSV:
+				fileDelimiter = "\t"
+			}
+		}
+
+		var output strings.Builder
+
+		switch format {
+		case FormatCSV:
+			// Use encoding/csv for proper CSV handling
+			writer := csv.NewWriter(&output)
+			writer.Comma = rune(fileDelimiter[0])
+
+			// Write headers
+			if err := writer.Write(nonEmptyHeaders); err != nil {
+				return "", fmt.Errorf("error writing CSV headers: %w", err)
+			}
+
+			// Write rows
+			for _, row := range rows {
+				csvRow := make([]string, len(nonEmptyColumnIndexes))
+				for j, i := range nonEmptyColumnIndexes {
+					csvRow[j] = row[i]
+				}
+				if err := writer.Write(csvRow); err != nil {
+					return "", fmt.Errorf("error writing CSV row: %w", err)
+				}
+			}
+			writer.Flush()
+			if err := writer.Error(); err != nil {
+				return "", fmt.Errorf("error flushing CSV writer: %w", err)
+			}
+
+		case FormatTSV:
+			// For TSV, replace tabs with spaces in values and use tab as delimiter
+			// Write headers
+			for i, header := range nonEmptyHeaders {
+				if i > 0 {
+					output.WriteString("\t")
+				}
+				output.WriteString(strings.ReplaceAll(header, "\t", " "))
+			}
+			output.WriteString(utils.GetLineEnding())
+
+			// Write rows
+			for _, row := range rows {
+				for j, i := range nonEmptyColumnIndexes {
+					if j > 0 {
+						output.WriteString("\t")
+					}
+					output.WriteString(strings.ReplaceAll(row[i], "\t", " "))
+				}
+				output.WriteString(utils.GetLineEnding())
+			}
+		}
+		return output.String(), nil
+
+	default:
+		// If format is empty, use table format for TTY and csv for non-TTY
+		if format == "" {
+			if term.IsTTYSupportForStdout() {
+				// Create a styled table for TTY
+				t := table.New().
+					Border(lipgloss.ThickBorder()).
+					BorderStyle(lipgloss.NewStyle().Foreground(lipgloss.Color(theme.ColorBorder))).
+					StyleFunc(func(row, col int) lipgloss.Style {
+						style := lipgloss.NewStyle().PaddingLeft(1).PaddingRight(1)
+						if row == -1 {
+							// Apply CommandName style to all header cells
+							return style.Inherit(theme.Styles.CommandName)
+						}
+						return style.Inherit(theme.Styles.Description)
+					}).
+					Headers(headers...).
+					Rows(rows...)
+
+				return t.String() + utils.GetLineEnding(), nil
+			} else {
+				// For non-TTY, use CSV format for better machine readability
+				format = FormatCSV
+				return FilterAndListStacks(stacksMap, component, listConfig, format, delimiter)
+			}
+		}
+
+		// For non-TTY or when format is explicitly "table", use consistent tabular format
+		// that matches the column configuration of the TTY output
+		var output strings.Builder
+
+		// Add a separator line after headers for better readability
+		headerLine := make([]string, len(headers))
```

Comment:
I am concerned that we are doing manual table generation even for plain text. There's gotta be either a charmbracelet or other library for this.